### DISCUSSION TITLE
Disabling Progress Bar in Explainers

### DISCUSSION THREAD

**Discussion by rglynn** [shap/shap#2161] [Q&A]
Is there currently no way to disable the progress bar for the explainers? The progress bar is nice in a jupyter notebook but if you run a similar mechanism in something like Airflow it makes the output log very messy. I don't currently see any option to disable the progress bar from showing.

**Comment by LetiP** [comment]
Yes, there is. Just use `explainer = shap.Explainer(model, silent=True)`.
Is this what you are looking for?

**Reply by marcoBmota8** [reply]
This does not work for the `shap.explainers.Tree` :$

**Reply by leonardomf12** [reply]
I believe you have to use the silent argument within the .shap_values() function and not directly on the Explainer.

shap_values = self.explainer.shap_values(x, silent=True)

**Comment by DanteTrb** [comment]
For general usage with `shap.Explainer`, you can disable the progress bar using the `silent=True` argument **inside the `.values()` or `.shap_values()` call**, not at the time of explainer instantiation.

 **Correct usage (TreeExplainer, etc.):**

```
explainer = shap.Explainer(model)
shap_values = explainer(X, silent=True)
```
❌ This will NOT work:

No effect  'silent' is not accepted here:
`explainer = shap.Explainer(model, silent=True)`

The silent parameter is passed during the computation of SHAP values, not when creating the explainer. This affects explainers like TreeExplainer which internally invoke C++/Cython-based logic that prints progress bars unless silent=True is set.

For Airflow, pipelines, or non-interactive environments, always prefer:
`shap_values = explainer(X, silent=True)`

This keeps logs clean and ensures compatibility in headless environments.

Hope this clarifies it! 🚀

**Reply by Gabriella-Chaos** [reply]
It seems like `silent` is not an argument for `.shap_values()` method. I got 
```
TypeError: TreeExplainer.shap_values() got an unexpected keyword argument 'silent'
```
